server/ai_worker.go

```diff
@@ -0,0 +1,532 @@
+					f, err := os.ReadFile(image.Url)
+					if err != nil {
+						clog.Errorf(ctx, "Could not create multipart part err=%q", err)
+						sendAIResult(ctx, n, orchAddr, notify.AIJobData.Pipeline, modelID, httpc, contentType, nil, addlResultData, err)
+						return
+					}
+					defer os.Remove(image.Url)
+					imgResp.Images[i].Url = fmt.Sprintf("%v.mp4", core.RandomManifestID())
+					w.SetBoundary(boundary)
+					hdrs := textproto.MIMEHeader{
+						"Content-Type":        {resultType},
+						"Content-Length":      {strconv.Itoa(len(f))},
+						"Content-Disposition": {"attachment; filename=" + imgResp.Images[i].Url},
+					}
+					fw, err := w.CreatePart(hdrs)
+					if err != nil {
+						clog.Errorf(ctx, "Could not create multipart part err=%q", err)
+						sendAIResult(ctx, n, orchAddr, notify.AIJobData.Pipeline, modelID, httpc, contentType, nil, addlResultData, err)
+						return
+					}
+					io.Copy(fw, bytes.NewBuffer(f))
+				}
+			}
+			//update resp for image.Url updates
+			resp = imgResp
+		}
+
+		//add the json to the response
+		//   audio-to-text has no file attachment because the response is json
+		jsonResp, err := json.Marshal(resp)
+
+		if err != nil {
+			clog.Errorf(ctx, "Could not marshal json response err=%q", err)
+			sendAIResult(ctx, n, orchAddr, notify.AIJobData.Pipeline, modelID, httpc, contentType, nil, addlResultData, err)
+			return
+		}
+
+		w.SetBoundary(boundary)
+		hdrs := textproto.MIMEHeader{
+			"Content-Type":   {"application/json"},
+			"Content-Length": {strconv.Itoa(len(jsonResp))},
+		}
+		fw, err := w.CreatePart(hdrs)
+		if err != nil {
+			clog.Errorf(ctx, "Could not create multipart part err=%q", err)
+		}
+		io.Copy(fw, bytes.NewBuffer(jsonResp))
+	}
+
+	w.Close()
+	contentType = "multipart/mixed; boundary=" + boundary
+	sendAIResult(ctx, n, orchAddr, notify.AIJobData.Pipeline, modelID, httpc, contentType, &body, addlResultData, nil)
+}
+
+func sendAIResult(ctx context.Context, n *core.LivepeerNode, orchAddr string, pipeline string, modelID string, httpc *http.Client,
+	contentType string, body *bytes.Buffer, addlData interface{}, err error,
+) {
+	taskId := clog.GetVal(ctx, "taskId")
+	clog.Infof(ctx, "sending results back to Orchestrator")
+	if err != nil {
+		clog.Errorf(ctx, "Unable to process AI job err=%q", err)
+		body.Write([]byte(err.Error()))
+		contentType = aiWorkerErrorMimeType
+	}
+	resultUrl := "https://" + orchAddr + "/aiResults"
+	req, err := http.NewRequest("POST", resultUrl, body)
+	if err != nil {
+		clog.Errorf(ctx, "Error posting results to orch=%s taskId=%d url=%s err=%q", orchAddr,
+			taskId, resultUrl, err)
+		return
+	}
+	req.Header.Set("Authorization", protoVerAIWorker)
+	req.Header.Set("Credentials", n.OrchSecret)
+	req.Header.Set("Content-Type", contentType)
+	req.Header.Set("TaskId", taskId)
+	req.Header.Set("Pipeline", pipeline)
+
+	//TODO consider adding additional information in response header from the addlData field (e.g. transcoding includes Pixels)
```

Comment:
@ad-astra-video related to the unsued arguments. If we need this information in the future we can add it. Please remove the TODO for now.